### PR TITLE
Command Additions

### PR DESCRIPTION
**Pull request by saladoc**
Includes
 - Index reordering (#202)
 - `CommandResult` explanation (#199)
 - `CommandException` explanation (#208)
 - Change preferred phase for command registration (adresses boformers comment on #197)

Removes
 - FLARD

### CODE REVIEW

File: source/plugin/basics/commands.rst [hunk 1]
```diff
@@ -340,14 +344,35 @@ Use the ``setChildren()`` method of the parent command builder to apply the chil
 
     If a ``CommandExecutor`` was set for the parent command, it is used as a fallback if the arguments do not match one of the child command aliases.
     Setting an executor is not required.
-    
+
+Command Results
+===============
+
+The ``CommandExecutor::execute()`` method must always return a ``CommandResult``. In most cases it is sufficient to return ``CommandResult.success()`` if the command was successful or ``CommandResult.empty()`` if it wasn't. 
+In cases where more information needs to be conveyed, a ``CommandResult.Builder`` should be used. It provides the methods ``affectedBlocks()``, ``affectedEntities()``, ``affectedItems()``, ``queryResult()`` and ``successCount()`` methods, each accepting an integer and setting the attribute of the same name. All attributes that are not set by the builder will be absent.
```
Comment (boformer) [inline source/plugin/basics/commands.rst]: CommandResult.builder()...

File: source/plugin/basics/commands.rst [hunk 2]
```diff
@@ -340,14 +344,35 @@ Use the ``setChildren()`` method of the parent command builder to apply the chil
 
     If a ``CommandExecutor`` was set for the parent command, it is used as a fallback if the arguments do not match one of the child command aliases.
     Setting an executor is not required.
-    
+
+Command Results
+===============
+
```
Comment (boformer) [inline source/plugin/basics/commands.rst]: Maybe a short explanation how the result is used, with link to minecraftwiki commandblock page

**Comment by boformer** [comment]
Maybe we should split up the page (like the block pages)? What do you think?

File: source/plugin/basics/commands.rst (hunk 1 above)
Comment (saladoc) [inline source/plugin/basics/commands.rst]: You're right, its simpler to just refer to the `builder()` method. Will change.

File: source/plugin/basics/commands.rst (hunk 2 above)
Comment (saladoc) [inline source/plugin/basics/commands.rst]: I'd have to understand it first, so dont expect it tomorrow morning ;)

Comment (boformer) [inline source/plugin/basics/commands.rst]: http://minecraft.gamepedia.com/Command_block#Output

> A command block can power a redstone comparator facing away from it (possibly separated by a block) with a signal strength specific to the last executed command (for example, the number of players affected by the command).
> The signal strength always reflects the last command executed, even after the command block is deactivated. 

and

> Command blocks can post various success results to scoreboard objectives, including:
> SuccessCount: the number of successes (previously also available by redstone comparator output, but not limited to 15)
> AffectedBlocks: the number of blocks that were altered
> AffectedEntities: the number of entities that were altered (including players)
> AffectedItems: the number of items that were altered
> QueryResult: value returned by command

**Comment by saladoc** [comment]
@boformer I already thought about that since the page has now quite large. I'll look into it.

**Comment by BlackGoat** [comment]
Looks OK, splitting the page was a good idea. Will there be any matching content on the page plugin/advanced/commands/arguments, or will that be dealt with separately?

**Comment by boformer** [comment]
There will not be any matching content on that page. It is a separate issue. 

We could merge the two "arguments" pages, so that only the CommandCallable page remains in the advanced section.

**Comment by BlackGoat** [comment]
Thanks. Do whatever makes the most sense to you. Someone let me know when this is ready to merge.

**Comment by boformer** [comment]
Except for the last commit, this is ready.

**Comment by saladoc** [comment]
That's why its the last commit, so it can manually be left out. My reasoning behind already including it was that 
 - probably most developers use the latest API snapshot anyway
 - the change is fully implemented in the API, so it will be available in any Sponge implementation that uses a SpongeAPI version newer than the linked commit. Which would be any current dev build.
 - if all docs had to be strictly 2.0, I'd have to change part of the block api docs since the `ImmutableDataHolder#withoutData()` method was introduced in SpongePowered/SpongeAPI@31467e86 - a 2.1 snapshot version.

As far as I am concerned, this can be merged. If strict compliance with SpongeAPI 2.0 is intended, just leave out the last commit.

**Comment by BlackGoat** [comment]
The dependencies page now demands Sponge API version 2.1-snapshot, so the rest of the docs must follow suit. Blame https://github.com/SpongePowered/SpongeDocs/commit/922ce29ae9336aa186a3730566779c81d8955b27

**Pull request merged**